### ISSUE TITLE
Leon uses current directory for temporary files

### ISSUE THREAD

**Issue by KirillKryukov** [GATB/leon#7]
Leon feels free to use current directory for storing temporary files.

For example, `leon -file 'data/1.fasta' -c -kmer-size 2` will create a temporary file `1.h5` in current directory. (Normally it's deleted after compression, but is left dangling if compression crashes).

Using current directory as temporary space is problematic because:

1. It's completely unexpected by the user.
1. The name may clash with other files already stored there.
1. Current directory might have insufficient space.
1. The process may have no write access to current directory.
1. When running concurrent leon tasks, these temporary files may clash with each other.

A slightly better solution would be to use output directory for temporary file storage (since we at least know that we have write access and most probably some free space there).

An even better way would be to use directory configured in TMPDIR environment variable. (Ideally with a command line option to specify another directory).

**Comment by rchikhi** [comment]
Thanks for the feedback. Output dir is a good idea, TMPDIR I'm not so sure as `/tmp` has often little space, and is even $TMPDIR set by all linux distros? (Not on my Ubuntu)

**Comment by KirillKryukov** [comment]
Yes, TMPDIR variable use is probably not universal. Also, on Windows, TMP and/or TEMP variables are used. So, perhaps safest is to default to using output directory.

In any case, adding any mechanism for specifying temporary directory will be an improvement compared to current state. Currently I have to chdir to a directory where I'd like to keep temporary files, then run leon, then chdir back.
